cvat/apps/engine/middleware.py

```diff
@@ -24,3 +28,27 @@ def __call__(self, request):
         response.headers["X-Request-Id"] = request.uuid
 
         return response
+
+
+class LastActivityMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        if request.user.is_authenticated:
+            profile = getattr(request.user, "profile", None)
+            if not profile:
+                return response
+
+            last_activity_date = profile.last_activity_date
+            if (
+                not last_activity_date
+                or (now() - last_activity_date) > settings.USER_LAST_ACTIVITY_UPDATE_MIN_INTERVAL
+            ):
+                # such way we avoid failing and any db updates if the Profile was removed during the request
+                Profile = apps.get_model("engine", "Profile")
```

Comment:
The `get_model` call seems unnecessary - why not just import `Profile` from `.models`?

Reply:
circular import

Reply:
You should be able to do a local import (just replace this line with an import statement).